lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/LibertyCompletionParticipant.java

```diff
@@ -53,12 +54,33 @@ public void onAttributeValue(String valuePrefix, ICompletionRequest request, ICo
         Properties variableProps = SettingsService.getInstance()
                 .getVariablesForServerXml(request.getXMLDocument()
                         .getDocumentURI());
-        String variableName = valuePrefix.replace("$", "")
-                .replace("{", "")
+        //getting all existing variables in current completion prefix string
+        List<VariableLoc> variables = LibertyUtils.getVariablesFromTextContent(valuePrefix);
+        // value prefix will contain all existing attribute value, till the cursor
+        String variableName = valuePrefix.replace("${", "")
                 .replace("}", "");
-        variableProps.entrySet().stream().filter(it -> it.getKey().toString().toLowerCase().contains(variableName.toLowerCase()))
+        StringBuilder prefixVariables = new StringBuilder();
+        String replacePrefix = "";
+        for (VariableLoc variableLoc : variables) {
+            prefixVariables.append(String.format("${%s}", variableLoc.getValue()));
```

Comment:
I don't understand this prefixVariables logic. It seems like you are iterating through all the variables found in the passed in valuePrefix and arbitrarily appending them to each other without regard to other characters that may have occurred between variable references in the original valuePrefix.

Reply:
I have modified the logic to support 4 scenarios

1. User already has multiple variables, added a new variable for completion
  a. User entered completion data  ${testVar1}/abc| -> in this case /abc is considered as completion variable and 
 probably there is no variable which contains /abc , hence no completion will be shown
  b. User entered completion data  ${testVar1}/${abc| -> in this case we identify the second variable as abc, tries to find any completion data which contains abc, and it will be shown with prefix ${testVar1}/
  c. User entered completion data  ${testVar1}/apps/${testVar2}/${abc|  -> same as case b, completion will be shown with prefix ${testVar1}/apps/${testVar2}/
  
  2. User do not have any existing variable in current attribute value
   a. User entered completion data "/apps" -> completion will be shown if there is any variable which contains "/apps"
   b. User entered completion data "/apps/${a" -> we identify ${ as starting of a variable, extract "a" as variable completion item, completes data based on any variable which contains prefix "/apps"